src/ios/MultipeerConnectivity/THEMultipeerClient.m

```diff
@@ -0,0 +1,284 @@
+
+    THEMultipeerClientSession *session = (THEMultipeerClientSession *)p;
+
+    NSString *peerIdentifier = info[PEER_IDENTIFIER_KEY];
+    if (session && ([[session remotePeerID] hash] == [peerID hash]) && 
+        [peerIdentifier isEqualToString:[session remotePeerIdentifier]])
+    {
+      NSLog(@"client: Found existing peer: %@", peerIdentifier);
+
+      clientSession = session;
+      previouslyVisible = clientSession.visible;
+    }
+    else
+    {
+      NSLog(@"client: Found new peer: %@", peerIdentifier);
+
+      // We've found a new peer, create a new record
+      clientSession = [[THEMultipeerClientSession alloc] 
+                                    initWithLocalPeerID:_localPeerId 
+                                       withRemotePeerID:peerID 
+                               withRemotePeerIdentifier:peerIdentifier];
+    }
+
+    [clientSession setVisible:YES];
+    return clientSession;
+  }];
+
+  if (clientSession && previouslyVisible == NO)
+  {
+    // A new peer or one that has become visible again. Only
+    // contact delegate when the state changes (we get duplicates a lot)
+    if ([_multipeerSessionDelegate respondsToSelector:@selector(didFindPeerIdentifier:peerName:)])
+    {
+      [_multipeerSessionDelegate 
+        didFindPeerIdentifier:[clientSession remotePeerIdentifier] 
+                     peerName:info[PEER_NAME_KEY]];
+    }
+  }
+}
+
+// Notifies the delegate that a peer was lost.
+- (void)browser:(MCNearbyServiceBrowser *)browser
+       lostPeer:(MCPeerID *)peerID
+{
+  __block BOOL previouslyVisible = NO;
+  __block THEMultipeerClientSession *clientSession = nil;
+
+  // Update the peer's record under lock
+  [_clientSessions updateForPeerID:peerID
+                       updateBlock:^THEMultipeerPeerSession *(THEMultipeerPeerSession *p) {
+
+    clientSession = (THEMultipeerClientSession *)p;
+    assert([[clientSession remotePeerID] hash] == [peerID hash]);
+
+    previouslyVisible = clientSession.visible;
+
+    if (clientSession)
+    {
+      NSLog(@"client: lost peer: %@", [clientSession remotePeerIdentifier]);
+      // disconnect will clear up any networking resources we currently hold
+      [clientSession disconnect];
+      [clientSession setVisible:NO];
+    }
+
+    return clientSession;
+  }];
+    
+  if (clientSession)
+  {
+    if (previouslyVisible == YES)
+    {
+      // Let interested parties know we lost a peer, only do this on a state change
+      if ([_multipeerSessionDelegate respondsToSelector:@selector(didLosePeerIdentifier:)])
+      {
+        [_multipeerSessionDelegate didLosePeerIdentifier:[clientSession remotePeerIdentifier]];
+      }
+    }
+  }
+  else
+  {
+    // Probably shouldn't happen
+    NSLog(@"WARNING: lostPeer we didn't know about");
+  }
+}
+
+- (void)browser:(MCNearbyServiceBrowser *)browser didNotStartBrowsingForPeers:(NSError *)error
+{
+  NSLog(@"WARNING: didNotStartBrowsingForPeers");
```

Comment:
Short answer: NO.

PeerID is not analogous to IP address. It's in the internals but there must be a per session id in there also. My evidence: You don't discover yourself normally, if you quickly stop and start broadcasting then you will see your previously advertised sessions. Nothing you provide to the API changes.

Seeing this log indicates programmer error and should really be an exception.